File: C/codeinginterview/offer_42/dp/Solution.c
//
// Created by 张智勇 on 2021/7/17.
//
int maxSubArray(int* nums, int numsSize) {
    int pre = 0, maxAns = nums[0];
    for (int i = 0; i < numsSize; i++) {
        pre = fmax(pre + nums[i], nums[i]);
        maxAns = fmax(maxAns, pre);
    }
    return maxAns;
}

File: Kotlin/src/codeinginterview/offer_42/dp/Solution.kt
package codeinginterview.offer_42.dp

class Solution {
    fun maxSubArray(nums: IntArray): Int {
        var pre = 0
        var maxAns = nums[0]
        for (x in nums) {
            pre = Math.max(pre + x, x)
            maxAns = Math.max(maxAns, pre)
        }
        return maxAns
    }
}